### PR TITLE
Upgrade to Gradle 8.8

### PR DESCRIPTION
**Pull request by jbonofre**
## Before this PR
Gradle 8.7 is used.

## After this PR
Upgrading to Gradle 8.8 to test with the target version.

## Possible downsides?
N/A

### CODE REVIEW

**Comment by jbonofre** [comment]
The `RevapiSpec.groovy` tests seem to be ok now here.

**Comment by jbonofre** [comment]
FYI, I found the root cause of the problem with Gradle 8.2+:
- with Gradle 8.1, the `oldJars` contains 1.5.0 artifacts (as defined in the `revapi` gradle configuration)
- with Gradle 8.2, the `oldJars` contains 1.6.0-SNAPSHOT (same as the `newJars`). That's why revapi doesn't detect any breaking change (it's comparing the same version/artifact).

The problem is located in the `ResolveOldApi` and `OldApiConfigurations` classes.

The `ResolveOldApi` cleanly defines the "old" version (`1.5.0` for instance), but it's changed in the `OldApiConfigurations.resolveOldConfiguration(project, groupNameVersion, false)` method. The problem is in this method in the way the version are resolved (related to the Gradle object). I'm fixing there.

I'm doing two things:
1. Fixing `RevolveOldApi`/`GradleUtils` for Gradle 8.2+
2. Refactoring cleaning the plugin to be Gradle 8.2+ friendly. For instance, the `RevapiAnalyzeTask` should be abstract, the getters should be abstract and we should use `ConfigurableFileCollection` instead of `Property<FileCollection>`.

@nastra ^^

File: build.gradle [hunk 1]
```diff
@@ -35,7 +35,7 @@ apply plugin: 'java-library'
 apply plugin: 'groovy'
 apply plugin: 'com.palantir.external-publish-jar'
 
-sourceCompatibility = 1.8
+sourceCompatibility = 11
```
Comment (nastra) [inline build.gradle]: I don't think we want to bumpt he JDK version. Otherwise we won't be able to use it in Iceberg

Comment (jbonofre) [reply build.gradle]: My bad, let me revert. Good catch !

**Comment by jbonofre** [comment]
@nastra new update: I identified the root cause of the problem (thanks to remote debugging).

The Gradle `Configuration` and `Version` passed to this method look correct: https://github.com/nastra/gradle-revapi/blob/develop/src/main/java/com/palantir/gradle/revapi/OldApiConfigurations.java#L53

However, when calling `configuration.resolve()` here https://github.com/nastra/gradle-revapi/blob/develop/src/main/java/com/palantir/gradle/revapi/OldApiConfigurations.java#L66 it's where the difference happens between Gradle 8.1 and 8.2+.

I'm checking the gap in `Configuration` in the two execution, and I'm checking if the `resolve()` method changed here.

**Comment by jbonofre** [comment]
I identified the gap. The problem is coming from `configuration.resolve()` with the following difference:
- with Gradle 8.1, `configuration` is `DefaultConfiguration` with a `resolver` field which is `ErrorHandlingConfigurationResolver` and delegate to `ShortCircuitEmptyConfigurationResolver`.
- with Gradle 8.8, `configuration` is `DefaultUnlockedConfiguration` with a `resolver` field which is directly `ShortCircuitEmptyConfigurationResolver`

I'm checking if I can change the resolver configuration.

**Comment by jbonofre** [comment]
I proposed a fix directly at palantir: https://github.com/palantir/gradle-revapi/pull/663

I'm including this change here as well.

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing this @jbonofre

**Pull request merged**